src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderParser.cs

```diff
@@ -1152,29 +1156,74 @@ private bool FetchNextField(ref ScanInfo scan, ReadOnlySpan<char> span)
                         ichCur++;
                         _sb.Clear();
                         int ichRun = ichCur;
-                        for (; ; ichCur++)
+                        if (_escapeChar == '"')
                         {
```

Comment:
Similar comment as the one in TextLoaderCursor. Please try to refactor if possible, these two if/else blocks.

Reply:
See my response above.

---
In reply to: [428387493](https://github.com/dotnet/machinelearning/pull/5147#discussion_r428387493) [](ancestors = 428387493)